src/ConnectedMode.UnitTests/Suppressions/ServerIssuesStoreTests.cs

```diff
@@ -30,6 +34,88 @@ public class ServerIssuesStoreTests
         public void MefCtor_CheckIsExported()
         {
             MefTestHelpers.CheckTypeCanBeImported<ServerIssuesStore, IServerIssuesStore>();
+            MefTestHelpers.CheckTypeCanBeImported<ServerIssuesStore, IServerIssuesStoreWriter>();
+        }
+
+        [TestMethod]
+        public void AddIssues_ListOfIssuesAsExpected()
+        {
+            var issue1 = CreateIssue("1", false);
+            var issue2 = CreateIssue("2", false);
+            var issue3 = CreateIssue("3", false);
+
+            var testSubject = new ServerIssuesStore();
+
+            var result = testSubject.Get();
+            result.Count().Should().Be(0);
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue1, issue2 }, clearAllExistingIssues: false);
+
+            result = testSubject.Get();
+            result.Count().Should().Be(2);
+            result.Should().Contain(issue1);
+            result.Should().Contain(issue2);
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue3 }, clearAllExistingIssues: true);
+
+            result = testSubject.Get();
+            result.Count().Should().Be(1);
+            result.Should().Contain(issue3);
+        }
+
+        [TestMethod]
+        public void UpdateIssues_ListOfIssuesAsExpected()
+        {
+            var issue1 = CreateIssue("issue1", true);
+            var issue2 = CreateIssue("issue2", false);
+
+            var testSubject = new ServerIssuesStore();
+
+            // If issue does not exist in list nothing should happen.
+            testSubject.UpdateIssue("issue1", false);
+            issue1.IsResolved.Should().BeTrue();
+
+            testSubject.AddIssues(new List<SonarQubeIssue>() { issue1, issue2 }, clearAllExistingIssues: false);
+
+            var result = testSubject.Get();
+            result.Count().Should().Be(2);
+            result.Should().Contain(issue1);
+            result.Should().Contain(issue2);
+            issue1.IsResolved.Should().BeTrue();
+            issue2.IsResolved.Should().BeFalse();
+
+            testSubject.UpdateIssue("issue1", false);
+            issue1.IsResolved.Should().BeFalse();
+
+            testSubject.UpdateIssue("issue2", true);
+            issue2.IsResolved.Should().BeTrue();
+
+            // This test is to insure it doesn't just flip the state.
+            testSubject.UpdateIssue("issue2", true);
+            issue2.IsResolved.Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void AddOrUpdateIssues_EventIsInvoked()
```

Comment:
Missing test: don't invoke the event if the issue key didn't match.

Reply:
(...which could be added in one of the new separation of tests I suggested above).